src/base/static/components/input-form/input-form.js

```diff
@@ -0,0 +1,833 @@
+            published={fieldValues[fieldConfig.name]}
+            onChange={evt => this.onFieldChange(evt, "value")}
+          />
+        );
+        break;
+      case constants.DATETIME_FIELD_TYPENAME:
+        return [
+          fieldPrompt,
+          <DatetimeField
+            name={fieldConfig.name}
+            date={fieldValues[fieldConfig.name]}
+            showTimeSelect={true}
+            onChange={(evt, name) => this.onValueNameFieldChange(evt, name)}
+          />
+        ];
+      case constants.GEOCODING_FIELD_TYPENAME:
+        return [
+          fieldPrompt,
+          <GeocodingField
+            onChange={evt => this.onFieldChange(evt, "value")}
+            name={fieldConfig.name}
+            value={fieldValues[fieldConfig.name]}
+            mapConfig={mapConfig}
+            emitter={emitter}
+          />
+        ];
+        break;
+      case constants.BIG_TOGGLE_FIELD_TYPENAME:
+        const values = [fieldConfig.content[0].value, fieldConfig.content[1].value];
+        return [
+          fieldPrompt,
+          <BigToggleField
+            name={fieldConfig.name}
+            checked={fieldValues[fieldConfig.name] === fieldConfig.content[0].value}
+            labels={[fieldConfig.content[0].label, fieldConfig.content[1].label]}
+            values={[fieldConfig.content[0].value, fieldConfig.content[1].value]}
+            id={"input-form-" + fieldConfig.name}
+            onChange={evt => this.onToggleFieldChange(evt, [fieldConfig.content[0].value, fieldConfig.content[1].value])}
+            autofillMode={autofillMode}
+            hasAutofill={fieldConfig.hasAutofill}
+          />
+        ];
+        break;
+      case constants.DROPDOWN_FIELD_TYPENAME:
+        return [
+          fieldPrompt,
+          <DropdownField
+            name={fieldConfig.name}
+            value={fieldValues[fieldConfig.name]}
+            options={fieldConfig.content}
+            onChange={evt => this.onFieldChange(evt, "value")}
+            autofillMode={autofillMode}
+            hasAutofill={fieldConfig.hasAutofill}
+          />
+        ];
+        break;
+      case constants.DROPDOWN_AUTOCOMPLETE_FIELD_TYPENAME:
+        return [
+          fieldPrompt,
+          <AutocompleteComboboxField
+            name={fieldConfig.name}
+            options={fieldConfig.content}
+            placeholder={fieldConfig.placeholder}
+            id={"autocomplete-" + fieldConfig.name}
+            onChange={this.onValueNameFieldChange.bind(this)}
+            showAllValues={true}
+            autofillMode={autofillMode}
+            hasAutofill={fieldConfig.hasAutofill}
+          />
+        ];
+        break;
+      case constants.COMMON_FORM_ELEMENT_TYPENAME:
+        const commonFormElementConfig = Object.assign(
+          {},
+          placeConfig.common_form_elements[fieldConfig.name],
+          {name: fieldConfig.name}
+        );
+        return this.buildField(commonFormElementConfig);
+        break;
+    }
```

Comment:
Is there a default case on this switch/break? We should at least throw an error if nothing else.